### PR TITLE
better debugging for webhooks

### PR DESCRIPTION
**Pull request by jonmoter**
Merge to master on Volunteer app wasn't triggering releases, but I have no idea why. Trying to add some better logging to hunt down what's up.

Still WIP.

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:

### Risks
- Level: Low/Med/High

### CODE REVIEW

File: app/controllers/integrations/base_controller.rb [hunk 1]
```diff
@@ -22,6 +23,20 @@ def create
...
   protected
 
+  # These methods can/must be overridden by subclasses
+
+  def validate_webhook
+    true # can be overridden in subclasses
+  end
+
+  def commit
+    raise NotImplementedError.new("#commit must be overridden in a subclass")
```
Comment (grosser) [inline app/controllers/integrations/base_controller.rb]: prefer `raise Xyz, message`

File: app/controllers/integrations/base_controller.rb [hunk 2]
```diff
@@ -22,6 +23,20 @@ def create
 
   protected
 
+  # These methods can/must be overridden by subclasses
+
+  def validate_webhook
```
Comment (grosser) [inline app/controllers/integrations/base_controller.rb]: would `validate_request` be better ?

File: app/controllers/integrations/github_controller.rb [hunk 3]
```diff
@@ -10,8 +10,12 @@ class Integrations::GithubController < Integrations::BaseController
 
   protected
 
+  def validate_webhook
+    head(:unauthorized) unless valid_signature?
```
Comment (grosser) [inline app/controllers/integrations/github_controller.rb]: maybe something more descriptive like `text: 'Invalid signature'`

**Comment by jonmoter** [comment]
@grosser now ready to merge

**Comment by grosser** [comment]
👍 so the problem was missing logs ?
... if anything else is fixed now can we add a test ?

**Comment by jonmoter** [comment]
No, this didn't actually fix the problem, just the first bit of refactoring to add debugging and try to figure out what went wrong. May add another log message or two before merging.

**Pull request merged**